odata-protocol/11.4 Data Modification.md

```diff
@@ -654,49 +668,72 @@ operations.
 
-Similarly, services MUST fail an update request to the URL of a [media entity](#RequestingtheMediaStreamofaMediaEntityusingvalue) that does not yet exist.
-However, a `PUT` request to the _media edit URL_ of a media entity does have Upsert
+Similarly, services MUST fail an upsert request to the URL of a [media entity](#RequestingtheMediaStreamofaMediaEntityusingvalue) that does not yet exist.
+However, a `PUT` request to the _media edit URL_ of a media entity does have upsert
 semantics, in that the media entity is [created](#CreateaMediaEntity)
 with the specified media stream if it does not already exist, otherwise the
 media stream of the existing media entity is
 [updated](#UpdateaMediaEntityStream).
 
-Singleton entities can be upserted if they are nullable. Services
-supporting this SHOULD advertise it by annotating the singleton with the
-term `Capabilities.UpdateRestrictions` (nested property `Upsertable`
-with value `true`) defined in [OData-VocCap](#ODataVocCap).
-
 A key property whose value is provided in the request URL SHOULD be omitted from the request body.
 If key properties are provided in the request URL and the request body with different values,
 services MUST either fail the request or ignore the value in the request body.
 
-To ensure that an update request is not treated as an insert, the client
-MAY specify an [`If-Match`](#HeaderIfMatch) header in the update
-request. The service MUST NOT treat an update request containing an
-`If-Match` header as an insert.
+If the URL of an upsert request falls into case 1, 2, or 3 given above and the request
+- contains an [`If-Match: *`](#HeaderIfMatch) header, the service MUST fail the request
+  if it would be treated as an insert
+- contains an `If-Match` header with an ETag as value, the service fails the request
+  instead of treating it as an insert because the [ETag](#UseofETagsforAvoidingUpdateConflicts)
+  does not match
+- contains an [`If-None-Match: *`](#HeaderIfNoneMatch) header, the service MUST fail the request
+  if it would be treated as an update.
 
-A `PUT` or `PATCH` request MUST NOT be treated as an update if an
-[`If-None-Match`](#HeaderIfNoneMatch) header is specified with a value
-of `*`.
+In other words, the `If-`(`None-`)`Match` header distinguishes between insert
+and update in these cases.
+
+Note outside of these cases it can happen that an upsert request does not address an existing entity with its URL
+but identifies one in its payload. Such a request may update the identified entity and
+link it to another entity, but it does not lead to a create entity request.
+
+::: example
+Example ##ex: `Category` is a single-valued nullable non-containment navigation property
+on the product entity type. Even if
+```
+GET http://host/service/Products(57)/Category
+```
+returns `404 Not Found`, the upsert request
+```
+PATCH http://host/service/Products(57)/Category
+```
+is treated like an update not an insert if its payload identifies an existing category.
```

Comment:
This PATCH request may or may not be treated as an insert, hence the `Upsertable` capability may or may not be relevant. This means that a client cannot know it advance whether it can make this request.